samples/grids/grid/editing-excel-style-custom/src/index.ts

```diff
@@ -0,0 +1,104 @@
+import 'igniteui-webcomponents-grids/grids/combined';
+import { ComponentRenderer, WebGridDescriptionModule } from 'igniteui-webcomponents-core';
+import { IgcCellType, IgcGridComponent } from 'igniteui-webcomponents-grids/grids';
+import { NwindData } from './NwindData';
+
+import "igniteui-webcomponents-grids/grids/themes/light/bootstrap.css";
+
+import "./index.css";
+
+export class Sample {
+
+    private grid1: IgcGridComponent
+    private _bind: () => void;
+
+    constructor() {
+        var grid1 = this.grid1 = document.getElementById('grid1') as IgcGridComponent;
+    
+        this._bind = () => {
+            grid1.data = this.nwindData;
+
+            grid1.addEventListener('activeNodeChange', (event: any) => {
+                grid1.endEdit();
+                grid1.markForCheck();
+
+            });
+            grid1.addEventListener('keydown', (event: KeyboardEvent) => {
+                var key = event.key; 
+                var activeElem = grid1.selectedCells[0];
+
+                if(key === 'Escape') {
+                    activeElem.editMode = false;
+                    return;
+                }
+
+                if ((key >= '0' && key <= '9') || (key.toLowerCase() >= 'a' && key.toLowerCase() <= 'z') && key != 'Enter') {
+        
+                    if (activeElem && activeElem.editMode === false) {
+                        activeElem.value = key;
```

Comment:
You should not set the value prop directly since this will automatically apply the change to the data. If you hit Escape after that to exit edit mode without commit of the change it will not revert it. 

The way you've set it up right now, if you type in "1234" on a cell it will set "1" as a value directly, so if you hit Escape the old value will not be set correctly. You will get "1", which is not correct.

Use `editValue` instead. You can set it after the cell enters edit mode.